standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/service/AcidHouseKeeperService.java

```diff
@@ -78,17 +81,13 @@ public void run() {
     }
   }
 
-  private void cleanTheHouse() {
+  protected void cleanTheHouse() {
     performTask(txnHandler::performTimeOuts, "Cleaning timed out txns and locks");
```

Comment:
could we refactor to
    protected final Map<FailableRunnable<MetaException>, String> tasks;
    in constructor:
    tasks = ImmutableMap.<FailableRunnable<MetaException>, String>builder()
      .put(txnHandler::performTimeOuts, "Cleaning timed out txns and locks")
      .put(txnHandler::cleanTxnToWriteIdTable, "Cleaning obsolete TXN_TO_WRITE_ID entries")
      .build();
    
tasks.forEach(this::performTask);

Reply:
corresponding change in Compaction service

Reply:
We cannot initialize tasks in constructor as we need txnHandler which is only initialized after setConf method is called. So made tasks as non final and initializing tasks in a new method